scheduler/job.go

```diff
@@ -0,0 +1,69 @@
+package scheduler
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/dealbot/tasks"
+	"github.com/robfig/cron/v3"
+)
+
+// Job is a set of parameters for running a job
+type Job struct {
```

Comment:
having 2 contexts on this struct feels awkward.

Reply:
Having one context in a struct is bad enough:
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

-- https://golang.org/pkg/context/

So, two cancels out, right?  Ok, maybe not so much.  I originally had put that together using channels, and it seemed even more awkward because of trying to use the context passed into the engine constructor.  Maybe the answer is to just ditch that context (use ctx for cancelation, to for lifecycle management) and this will all clear up.  I will take another pass at that.

Reply:
Got rid of all contexts in structs.  Ends up being a much nicer solution.